msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AcquireTokenByAuthorizationGrantSupplier.java

```diff
@@ -77,55 +77,56 @@ private boolean IsUiRequiredCacheSupported() {
     private OAuthAuthorizationGrant processPasswordGrant(
             OAuthAuthorizationGrant authGrant) throws Exception {
 
-        if (!(authGrant.getAuthorizationGrant() instanceof ResourceOwnerPasswordCredentialsGrant)) {
+        if (!(authGrant.toParameters().get(GrantConstants.GRANT_TYPE_PARAMETER).get(0).equals(GrantConstants.PASSWORD))) {
             return authGrant;
         }
 
         if (msalRequest.application().authenticationAuthority.authorityType != AuthorityType.AAD) {
             return authGrant;
         }
 
-        ResourceOwnerPasswordCredentialsGrant grant =
-                (ResourceOwnerPasswordCredentialsGrant) authGrant.getAuthorizationGrant();
-
         UserDiscoveryResponse userDiscoveryResponse = UserDiscoveryRequest.execute(
-                this.clientApplication.authenticationAuthority.getUserRealmEndpoint(grant.getUsername()),
+                this.clientApplication.authenticationAuthority.getUserRealmEndpoint(authGrant.toParameters().get("username").get(0)),
                 msalRequest.headers().getReadonlyHeaderMap(),
                 msalRequest.requestContext(),
                 this.clientApplication.serviceBundle());
 
         if (userDiscoveryResponse.isAccountFederated()) {
             WSTrustResponse response = WSTrustRequest.execute(
                     userDiscoveryResponse.federationMetadataUrl(),
-                    grant.getUsername(),
-                    grant.getPassword().getValue(),
+                    authGrant.toParameters().get(GrantConstants.USERNAME_PARAMETER).get(0),
+                    authGrant.toParameters().get(GrantConstants.PASSWORD_PARAMETER).get(0),
                     userDiscoveryResponse.cloudAudienceUrn(),
                     msalRequest.requestContext(),
                     this.clientApplication.serviceBundle(),
                     this.clientApplication.logPii());
 
-            AuthorizationGrant updatedGrant = getSAMLAuthorizationGrant(response);
+            Map<String, List<String>> params = getSAMLAuthGrantParameters(response);
+            params.putAll(authGrant.toParameters());
 
-            authGrant = new OAuthAuthorizationGrant(updatedGrant, authGrant.getParameters());
+            authGrant = new OAuthAuthorizationGrant(params, null, null);
```

Comment:
Maybe some constructor overloads if `null, null` is a commonish case.

Reply:
This spot and one other in a test were the only places that used a double null, however another also had a null for the third parameter (claims) but not the second (scopes).

In the latest commit I added a constructor that omits the claims parameter, as that's the one that's most likely to be null in a flow.